ApollosProject/apollos-shovel
# Contentful Sync (Kids Club Data Only)

### About

This PR adds the sync capability to our contentful integration and creates the sync for the Crossroads Kids Club Data.

We're now saving sync tokens in redis. Kids club sync tokens are saved under the key `crossroads_kids_club_contentful_sync_token`. I did not use the redis hook that airflow built because it's surprisingly limited. Instead, we're using the regular ol' `redis` python package. The redis connection URL is now stored as an airflow variable.

To get the sync working, I had to tweak the original backfill. We had been backfilling video content only, but that led to issues during the sync. When you specify a content type (in this case, we specified that we were backfilling `Entries` of type `Video`), all following syncs also only use that specific type. In order to get updates to assets and categories (in the future), I had to open up the backfill to sync _all_ content, not just entries of type `Video`.

I definitely wasn't able to test all of the sync scenarios, so we'll watch this run over the next few days to make sure its behaving as expected. 

### Testing

1. Start up Airflow.
2. Toggle on the `crossroads_kids_club_backfill_contentful_content_item_dag` DAG. Wait for it to finish, it will take a few minutes.
3. Toggle on the `crossroads_kids_club_contentful_content_item_dag`, and wait for it to finish. It should run much faster 😅

### Screenshots
 
<img width="760" alt="Screen Shot 2022-01-17 at 6 07 10 PM" src="https://user-images.githubusercontent.com/34075293/149846109-5b667f2a-5182-4e80-81a7-5940b2874253.png">
